util/codec/bytes.go

```diff
@@ -47,9 +47,12 @@ import (
 // EncodeBytes guarantees the encoded value is in ascending order for comparison,
 // The encoded value is >= SmallestNoneNilValue and < InfiniteValue.
 func EncodeBytes(b []byte, data []byte) []byte {
+	// Allocate more space to avoid unnecessary slice growing
+	bs := make([]byte, len(b), len(b)+len(data)+20)
```

Comment:
sometimes we will alloc a proper b outer and reuse it to reduce memory allocation. like
```
b := make([]byte, 0, 1024)
b = EncodeBytes(b, data1)
b = EncodeBytes(b, data2)
```

I think this is more efficient than allocing inner each time.